File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.ts
function maxScoreSightseeingPair(values: number[]): number {
    let maxScope:number = 0;
    let maxLeft:number = values[0];

    for(let i=1;i<values.length; i++){
        maxScope = Math.max(maxScope,maxLeft+values[i]-i);
        maxLeft = Math.max(maxLeft, values[i]+i);
    }
    return maxScope;
};

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.py
class Solution(object):
    def maxScoreSightseeingPair(self, values):
        """
        :type values: List[int]
        :rtype: int
        """
        maxScope = 0
        maxLeft = values[0]
        for i in range(1,len(values)):
            maxScope = max(maxScope,maxLeft+values[i]-i)
            maxLeft = max(maxLeft,values[i]+i)
        
        return maxScope